### DISCUSSION TITLE
Losing Full-Row Width Tabulator

### DISCUSSION THREAD

**Discussion by kushfest** [anvilistas/anvil-extras#335] [Q&A] [answered]
Hi All,

I'm using the navigation module and Tabulator in an app and I'm having a strange issue where the form with the tabulator panel loads correctly when I navigate to it the first time (ie. the tabulator panel is full row width). However, when I navigate away (ie. go to the Home panel) and return to that form, suddenly the Tabulator panel itself is left justified and not full-width, the panel enclosing it however is still full-width. 

[Here](https://anvil.works/build#clone:HIZXPLXCWZW7LSC7=K4A3XNC7XULLUHM7EXPSBF6W|C6ZZPAPN4YYF5NVJ=|TGQCF3WT6FVL2EM2=) is a minimal example clone link that demonstrates the issue.

@s-cork was generous enough to point out my obvious miss to get full_width working with the navigation panel in the first place and I hope/suspect I'm missing something painfully obvious again. Any help would be greatly appreciated!

**Comment by s-cork** [comment, marked as answer]
Thanks for the minimal example.
This behaviour is because of the way you're setting the default tabulator options
Setting the `Tabulator.default_options` inside the init method isn't the right place for it.
These should be defined outside of the class definition.
They're effectively global options and should be set one time.

the `fitDataTable` option does what it says - it fits the table to the data.
You only see this on the second load.
It doesn't get applied the first load because the the `Tabulator.default_options` were set after the `tabulator` component was created.
the second time the page loads the default_options are now set so you're seeing the `tabulator` component with the `fitDataTable` layout.
_(the first time it loads you get the `fitColumns` layout)_

**Reply by kushfest** [reply]
Stu! Thank you again for very quickly pointing out the very obvious mistake I was making (and nowhere near figuring out) and taking the time to help me understand. It's working great now!
